tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/AbstractTychoMapping.java

```diff
@@ -261,18 +255,20 @@ protected Charset getPrimaryArtifactCharset() {
         return StandardCharsets.UTF_8;
     }
 
-    protected abstract void initModel(Model model, Reader artifactReader, File artifactFile)
-            throws ModelParseException, IOException;
+    protected abstract void initModel(Model model, Reader artifactReader, File artifactFile) throws IOException;
 
     protected static Properties getBuildProperties(File dir) throws IOException {
-        Properties properties = new Properties();
-        File buildProperties = new File(dir, "build.properties");
-        if (buildProperties.exists()) {
-            try (FileInputStream stream = new FileInputStream(buildProperties)) {
+        Properties buildProperties = new Properties();
+        loadProperties(new File(dir, "build.properties"), buildProperties);
+        return buildProperties;
+    }
+
+    static void loadProperties(File propertiesPath, Properties properties) throws IOException {
```

Comment:
On quick glance it looks like you could always create the properties inside, since the input is always empty. I might be wrong, however.

Reply:
In `TychoBundleMapping.initModel()` this is called multiple times with the same properties object in order to avoid multiple loads of the same file for each localized property.
However I reworked that part of the code and this method now creates and returns a Properties object on each invocation.